### PR TITLE
add timer save request in data_collection_server

### PR DESCRIPTION
**Pull request by knorth55**
- add `timer` method for data collection
  - `start_request` service to start save timer
  - `end_request` service to end save timer

- add `method` param to select `request`, `message_filters` and `timer`
  - add deprecated warning for `with_request`

### CODE REVIEW

**Comment by wkentaro** [comment]
`start_request` and `end_request` is useful for `message_filter`, so maybe you need to change the design of condition using rosparam.

File: jsk_data/node_scripts/data_collection_server.py [hunk 1]
```diff
@@ -83,19 +83,42 @@ def __init__(self):
                     jsk_logfatal("Required field '{}' for param is missing"
                                  .format(field))
                     sys.exit(1)
-        if rospy.get_param('~with_request', True):
+        params = rospy.get_param_names()
+        method = rospy.get_param('~method', 'request')
+        if rospy.get_name() + '/with_request' in params:
```
Comment (wkentaro) [inline jsk_data/node_scripts/data_collection_server.py]: rospy.has_param?

Comment (knorth55) [reply jsk_data/node_scripts/data_collection_server.py]: good! thx

**Comment by knorth55** [comment]
> start_request and end_request is useful for message_filter, so maybe you need to change the design of condition using rosparam.

What do you mean?

**Comment by wkentaro** [comment]
You may want to use `start_request` and `end_request` with `_method:=message_filter` in the future, but current implementation cannot do that.
I mean there is no strong restraints between `start/end_request` and `_method:=message_filter`.

**Comment by k-okada** [comment]
@knorth55 kindly ping, I want this in next release

**Comment by furushchev** [comment]
@k-okada 

@knorth55 said they need time for addressing this pull request. How about including this into next to the next release?

**Comment by furushchev** [comment]
@k-okada ping

**Comment by k-okada** [comment]
@knorth55 Is 1 month enough for addressing this issue ???

**Comment by knorth55** [comment]
@k-okada I updated to support `message_filters`.

**Comment by knorth55** [comment]
@wkentaro Can you review again this?

**Comment by wkentaro** [comment]
Could you please check https://github.com/knorth55/jsk_common/pull/1?

**Comment by knorth55** [comment]
I checked your PR.
sync is correct position because callback function differs in each method.

**Pull request merged**